### ISSUE TITLE
`inactivity_period` in `okta_app_signon_policy_rule` should not default to `PT1H`

### ISSUE THREAD

**Issue by chickenriggies** [okta/terraform-provider-okta#1573]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.4.5
on darwin_amd64
+ provider registry.terraform.io/okta/okta v4.0.0

### Affected Resource(s)

- okta_app_signon_policy_rule

### Terraform Configuration Files

```hcl
resource "okta_app_signon_policy_rule" "fido" {
  name                                   = "fido"
  policy_id                              = okta_app_signon_policy.fido.id
  access                                 = "ALLOW"
  factor_mode                            = "2FA"
  groups_included                        = [
    local.groups["Linux Users"].id
  ]
  network_connection                     = "ANYWHERE"
  priority                               = 98
  re_authentication_frequency            = "PT43800H"
  constraints                            = [
    jsonencode({
      "possession" : {
        "types" : [
          "security_key"
        ],
        "methods" : [
          "webauthn"
        ]
      }
    })
  ]
}
```

### Debug Output

https://gist.github.com/hb3b/14259a23846b15ce848288e832845bdd

### Expected Behavior

`inactivity_period` should not be set unless deliberately set by the user.

#### Can this be done in the Admin UI?

`inactivity_period` does not appear to be configurable in the Admin UI.

#### Can this be done in the actual API call?

Yes, through omission.

### Actual Behavior

`inactivity_period` is set to a default value of: `PT1H`.

### References

- https://github.com/okta/terraform-provider-okta/pull/1184
- https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_signon_policy_rule#inactivity_period

**Comment by cjh-Ella** [comment]
Hello,

we encountered the same issue. Can we reopen this issue?

Thank you!

**Comment by pro4tlzz** [comment]
For now you can ignore this specific attribute, I too ran into this issue

**Comment by duytiennguyen-okta** [comment]
https://oktainc.atlassian.net/browse/OKTA-728926

**Comment by jbow-square** [comment]
This issue is still persistant in the current provider

**Comment by MichalMMac** [comment]
Update: This problem was introduced in v5.0.0.
**Update 2: I can use `inactivity_period = null`. This gets applied successfully.**

Hey. I am a trying to upgrade the provider from `4.18.0` to `6.2.0` but I run into an issue with `inactivity_period`.

With 4.18.0 I could set `inactivity_period` to null by doing this `inactivity_period = ""`.
However when I try to use `inactivity_period = ""` with 6.2.0, `terraform plan/apply` **always** produces this change:

```
  # okta_app_signon_policy_rule.test-auth-policy-rule will be updated in-place
  ~ resource "okta_app_signon_policy_rule" "test-auth-policy-rule" {
        id                          = "SOME_ID"
      - inactivity_period           = "PT1H" -> null
        name                        = "Test"
        # (15 unchanged attributes hidden)
    }
```

If I don't explicitly set `inactivity_period` the default value `PT1H` is used.

```
>>terraform state show okta_app_signon_policy_rule.test-auth-policy-rule 
# okta_app_signon_policy_rule.test-auth-policy-rule:
resource "okta_app_signon_policy_rule" "test-auth-policy-rule" {
....
    inactivity_period           = "PT1H"
...
}
```

**Comment by NMackert** [comment]
From what I saw when upgrading to 6.5.5, all our policies without explicitly set inactivity_period have changed to default to null. So I believe this was fixed?

**Comment by aditya-okta** [comment]
@NMackert , yes this was fixed by [2614](https://github.com/okta/terraform-provider-okta/pull/2614).Thanks.
